src/Ng/CommandHelpers.cs

```diff
@@ -45,275 +55,269 @@ private static void TraceRequiredArgument(string name)
             Trace.TraceError("Required argument \"{0}\" not provided", name);
         }
 
-        public static string GetSource(IDictionary<string, string> arguments)
+        private static bool TryGetArgument(IDictionary<string, string> arguments, string searchArg, out string value, bool required = false)
         {
-            string value;
-            if (!arguments.TryGetValue("-source", out value))
+            if (!arguments.TryGetValue(Constants.ArgumentPrefix + searchArg, out value))
             {
-                TraceRequiredArgument("-source");
-                return null;
+                if (required)
+                {
+                    TraceRequiredArgument(searchArg);
+                    throw new ArgumentException("Required argument not provided", searchArg);
+                }
+
+                value = null;
+                return false;
+            }
+
+            return true;
+        }
+
+        public static SecretInjector GetSecretInjector(IDictionary<string, string> arguments)
+        {
+            ISecretReader secretReader;
+            try
+            {
+                string vaultName;
+                if (!TryGetArgument(arguments, Constants.VaultName, out vaultName))
+                {
+                    secretReader = new EmptySecretReader();
+                }
+                else
+                {
+                    string clientId;
+                    TryGetArgument(arguments, Constants.ClientId, out clientId, required: true);
+
+                    string certificateThumbprint;
+                    TryGetArgument(arguments, Constants.CertificateThumbprint, out certificateThumbprint, required: true);
+
+                    bool shouldValidateCertificate = GetBool(arguments, Constants.ValidateCertificate, defaultValue: false);
+
+                    var keyVaultConfiguration = new KeyVaultConfiguration(vaultName, clientId, certificateThumbprint, shouldValidateCertificate);
+                    secretReader = new KeyVaultReader(keyVaultConfiguration);
+                }
+            }
+            catch (ArgumentException)
+            {
+                secretReader = new EmptySecretReader();
             }
```

Comment:
this is still not the desired behavior, since this method never fails. If VaultName is provided, but ClientId is not, we will succeed and return EmptySecretReader to the caller, instead of failing all together and printing usage help string.